test/recommend/test_recommend_pipeline.py

```
@@ -152,20 +153,27 @@ def test_list_dir(self):
 
     def test_xml(self):
         mock_input = self.__create_mock_input()
         mock_cmd = self.__create_mock_cmd()
         root = create_xml_tree(mock_input, full_report=True, cmd=mock_cmd, exit_code=2)
+        validate_xml_report(root)
 
         self.assertEqual(root.findtext('./header/patterns'), '5')
         self.assertEqual(len(root.findall('./files/file/patterns/pattern')), 5)
 
+    def test_xml_validation_rejects_invalid_report(self):
+        with self.assertRaises(Exception):
```

Ruff (0.15.21)

[warning] Do not assert blind exception: `Exception`

(B017)